quaterion_models/heads/skip_connection_head.py

```diff
@@ -34,22 +34,38 @@ class SkipConnectionHead(EncoderHead):
         skip_dropout:
             Additional dropout, applied to the trainable branch only.
             Using additional dropout allows to avoid the modification of original embedding.
+        n_layers:
+            Number of gated residual blocks stacked on top of each other.
     """
 
     def __init__(
-        self, input_embedding_size: int, dropout: float = 0.0, skip_dropout: float = 0.0
+        self,
+        input_embedding_size: int,
+        dropout: float = 0.0,
+        skip_dropout: float = 0.0,
+        n_layers: int = 1,
     ):
         super().__init__(input_embedding_size, dropout=dropout)
-        self.gates = Parameter(torch.Tensor(self.input_embedding_size))
-        self.reset_parameters()
-        self.fc = torch.nn.Linear(input_embedding_size, input_embedding_size)
-        self._skip_dropout_prob = skip_dropout
+        for i in range(n_layers):
+            setattr(
+                self, f"gates_{i}", Parameter(torch.Tensor(self.input_embedding_size))
+            )
+            setattr(
+                self,
+                f"fc_{i}",
+                torch.nn.Linear(input_embedding_size, input_embedding_size),
```

Comment:
https://discuss.pytorch.org/t/nn-module-add-new-parameter-setattr-vs-register-parameter/108238 - might it be relevant?

Reply:
Updated. Now with `n_layers=2`, `head.state_dict().keys()` gives the following: `odict_keys(['gates_0', 'gates_1', 'fc_0.weight', 'fc_0.bias', 'fc_1.weight', 'fc_1.bias'])`